WordPress/Classes/Services/NotificationFilteringService.swift

```diff
@@ -50,9 +54,21 @@ final class NotificationFilteringService {
     }
 
     func shouldFilterWordPressNotifications() -> Bool {
-        return allowDisablingWPNotifications
+        let shouldFilter = allowDisablingWPNotifications
             && isWordPress
             && !wordPressNotificationsEnabled
+
+        if shouldFilter {
+            cancelAllPendingWordPressLocalNotifications()
```

Comment:
IMHO, we should avoid including side effects in such a method. 

Such usage is more transparent:
```
if shouldFilterWordPressNotifications() {
    cancelAllPendingWordPressLocalNotifications()
}
```

Reply:
@staskus thanks for your comment.  I had done it the way you suggested at first but this results in repeat calls to `cancelAllPendingWordPressLocalNotifications()` each time we check `shouldFilterWordPressNotifications()`.  At present this would be duplicated twice re: current usage in code base.  As a next step I suggest we update the function name?  

As a suggestion how about something like the pseudo code below to indicate that there will be an additional operation?

`func shouldFilterWordPressNotifications(shouldCancelPendingNotifications: Bool = true)`

Thanks!

Reply:
> I had done it the way you suggested at first but this results in repeat calls to cancelAllPendingWordPressLocalNotifications() each time we check shouldFilterWordPressNotifications()

Okay, I understand the problem now, thanks for explaining! 👍 

We want to cancel pending local notifications every time we ask `shouldFilterWordPressNotifications`, **in case** there're pending notifications. Thanks for explaining. 

Although it's not ideal, I do agree then that it's better to have this logic in one place, instead of scattered in different places where we have this check. Especially given that it's a temporary feature that needs to be removed in a couple of months. I don't even think it's worth moving this decision to the caller then and pass a `boolean`.

Reply:
🙇